tests/unit_tests/test_league_rater.py

```diff
@@ -0,0 +1,363 @@
+    current_score = LeagueScore(division_id=999, score=4, game_count=config.PLACEMENT_GAMES)
+    rating = (150.0 - config.RATING_MODIFIER_FOR_PLACEMENT, 0.0)
+
+    new_score = LeagueRater.rate(example_league, current_score, None, rating)
+
+    assert new_score.division_id == example_league.divisions[1].id
+    assert new_score.game_count == current_score.game_count + 1
+    assert new_score.score == 5
+
+
+def test_placement(example_league, unplaced_player_score):
+    # Neutralize the offset in the placement function so we can test the score independently of the config settings
+    rating = 150 - config.RATING_MODIFIER_FOR_PLACEMENT
+
+    new_score = LeagueRater._do_placement(example_league, unplaced_player_score, rating)
+
+    assert new_score.division_id == example_league.divisions[1].id
+    assert new_score.game_count == unplaced_player_score.game_count + 1
+    assert new_score.score == 5
+
+
+def test_placement_high_rating(example_league, unplaced_player_score):
+    rating = 1500 - config.RATING_MODIFIER_FOR_PLACEMENT
+
+    new_score = LeagueRater._do_placement(example_league, unplaced_player_score, rating)
+
+    assert new_score.division_id == example_league.divisions[-1].id
+    assert new_score.game_count == unplaced_player_score.game_count + 1
+    assert new_score.score == 10
+
+
+def test_placement_low_rating(example_league, unplaced_player_score):
+    rating = -500 - config.RATING_MODIFIER_FOR_PLACEMENT
+
+    new_score = LeagueRater._do_placement(example_league, unplaced_player_score, rating)
+
+    assert new_score.division_id == example_league.divisions[0].id
+    assert new_score.game_count == unplaced_player_score.game_count + 1
+    assert new_score.score == 0
+
+
+def test_new_player(example_league):
+    current_score = LeagueScore(division_id=None, score=None, game_count=0)
+    player_rating = (380.0, 0.0)
+
+    new_score = LeagueRater.rate(
+        example_league,
+        current_score,
+        GameOutcome.VICTORY,
+        player_rating,
+    )
+
+    assert new_score.division_id == current_score.division_id
+    assert new_score.game_count == 1
+    assert new_score.score == current_score.score
+
+
+def test_placement_games(example_league):
+    current_score = LeagueScore(division_id=None, score=None, game_count=5)
+    player_rating = (380.0, 0.0)
+
+    new_score = LeagueRater.rate(
+        example_league,
+        current_score,
+        GameOutcome.VICTORY,
+        player_rating,
+    )
+
+    assert new_score.division_id == current_score.division_id
+    assert new_score.game_count == 6
+    assert new_score.score == current_score.score
+
+
+def test_promote(example_league):
+    current_score = LeagueScore(division_id=2, score=10, game_count=30)
+    player_rating = (380.0, 0.0)
+
+    new_score = LeagueRater.rate(
+        example_league,
+        current_score,
+        GameOutcome.VICTORY,
+        player_rating,
+    )
+
+    assert new_score.division_id == 3
```

Comment:
This one is a bit awkward but maybe expresses better what you want to test?
```suggestion
    assert new_score.division_id == example_league.get_next_higher_division(current_score.division_id).id
```

Reply:
I'd rather have the explicit result here. Someone might manage to fuck up the get_next_higher_division method in a way that the test still passes if we use the same method to confirm the test result.